src/ralph/backends/data/fs.py

```diff
@@ -184,30 +183,15 @@ def read(  # noqa: PLR0913
             BackendException: If a failure during the read operation occurs and
                 `ignore_errors` is set to `False`.
         """
-        if not query.query_string:
-            query.query_string = self.default_query_string
-
         if not chunk_size:
             chunk_size = self.default_chunk_size
 
-        target = Path(target) if target else self.default_directory
-        if not target.is_absolute() and target != self.default_directory:
-            target = self.default_directory / target
-        paths = list(
-            filter(lambda path: path.is_file(), target.glob(query.query_string))
-        )
-
-        if not paths:
-            logger.info("No file found for query: %s", target / query.query_string)
-            return
-
-        logger.debug("Reading matching files: %s", paths)
-
-        for path in paths:
-            with path.open("rb") as file:
-                reader = self._read_raw if raw_output else self._read_dict
-                for chunk in reader(file, chunk_size, ignore_errors):
+        for file, path in self._iter_files_matching_query(target, query):
+            if raw_output:
+                while chunk := file.read(chunk_size):
```

Comment:
If I understand correctly, the chunk size here is in bytes and not in number of documents as it is the case with other backends. Is this correct ? If yes, shouldn't we change this ?

Either case, I think the behavior of `chunk_size` (bytes or documents) should be explicited in the docstring.

Reply:
Good Idea) Thanks) added more docstring explication in PR #519 (dedicated to the `chunk_size` argument)
```
chunk_size (int or None): The chunk size when reading files.
                If `chunk_size` is `None` it defaults to `READ_CHUNK_SIZE`.
                If `raw_output` is set to `False`, files are read line by line.
```